public/js/deleteCategory.js

```diff
@@ -12,10 +13,13 @@ deleteButtons.forEach(deleteCatBtn => {
                 }
             });
             const responseData = await response.json();
-            document.cookie = `response = ${responseData.message}`;
-            window.location.reload();
+            if (responseData.success) {
+                window.location.reload();
+            } else {
+                deleteError.classList.remove('hidden');
```

Comment:
the response should have an error message in it that says what went wrong, perhaps display that error here rather than a generic one?